src/main/java/nextstep/path/application/ShortestPathFinder.java

```diff
@@ -0,0 +1,50 @@
+package nextstep.path.application;
+
+import nextstep.exceptions.ErrorMessage;
+import nextstep.path.domain.WeightedEdge;
+import nextstep.path.exceptions.PathNotFoundException;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Component
+public class ShortestPathFinder<EDGE extends WeightedEdge<VERTEX>, VERTEX> {
+    private final WeightedMultigraph<VERTEX, DefaultWeightedEdge> graph;
+    private final DijkstraShortestPath<VERTEX, DefaultWeightedEdge> shortestPath;
+
+    public ShortestPathFinder() {
+        graph = new WeightedMultigraph<>(DefaultWeightedEdge.class);
+        shortestPath = new DijkstraShortestPath<>(graph);
+    }
+
+    public Optional<GraphPath<VERTEX, DefaultWeightedEdge>> find(final List<EDGE> edges, VERTEX source, VERTEX target) {
+        addEdges(edges);
+        setWeight(edges);
+        try {
+            return Optional.ofNullable(shortestPath.getPath(source, target));
+        }catch (IllegalArgumentException exception) {
```

Comment:
코드 포맷팅 확인해주시면 좋을 것 같습니다.
```suggestion
        } catch (IllegalArgumentException exception) {
```